---
title: nf-core
description: A community effort to collect a curated set of analysis pipelines built using Nextflow
projectURL: https://nf-co.re
github: nf-core
# heroImage: /post_img.webp
iconImage: /images/projects/nf-core-logo-square.svg
iconImageDark: /images/projects/nf-core-logo-square-dark.svg
logoImage: /images/projects/nf-core-logo.svg
logoImageDark: /images/projects/nf-core-logo-darkbg.svg
order: 20
---

import TimeLineElement from "@components/TimeLine.astro";

[nf-core](https://nf-co.re) is a collaborative, community-driven project that provides a collection of high-quality, peer-reviewed bioinformatics pipelines built using Nextflow. The project was born out of the need to standardise workflows across different research groups and make complex bioinformatics analyses more accessible and reproducible.

I started nf-core while working at the National Genomics Infrastructure (NGI) in SciLifeLab, Sweden. At the time, I saw firsthand how teams often developed their own pipelines independently, leading to a lack of consistency and reproducibility across labs. There was a clear need for a shared resource where bioinformaticians could work together to build pipelines that were robust, scalable, and well-documented.

With nf-core, the goal is to enable collaboration and bring the bioinformatics community together to create workflows that everyone can benefit from. By sharing knowledge and pooling resources, we can build better pipelines—pipelines that adhere to strict standards and are rigorously tested across multiple platforms to ensure they are reliable.

nf-core pipelines are designed to:

- Provide **standardised workflows** for a wide range of bioinformatics tasks, from RNA-seq to epigenomics.
- Ensure **reproducibility** with locked software versions and containerised dependencies using Docker, Singularity, or Conda.
- **Scale** from local machines to high-performance clusters or cloud platforms without needing modification.
- Encourage **collaboration** by allowing anyone to contribute, review, and improve pipelines, ensuring they stay up to date and fit the needs of the community.

The project places a strong emphasis on **collaboration and openness**. All nf-core pipelines undergo peer review and continuous testing to ensure they work reliably across different environments. This makes nf-core an essential resource for researchers looking to perform reproducible, scalable bioinformatics analyses while contributing to the community at large.

Starting nf-core was about more than just making pipelines – it was about building a collaborative, supportive community around them. Today, it’s used globally by research groups who benefit from this shared resource, ensuring their analyses are not only efficient but also reliable and reproducible.

The source code and documentation for all nf-core pipelines are available on GitHub at [nf-core GitHub](https://github.com/nf-core), where everyone is welcome to contribute.

{/* prettier-ignore */}
<div class="time-line-container clear-both mb-10 grid gap-4">
  <TimeLineElement title="Reached 10k members" subtitle="September 2024">
    nf-core Slack reaches 10,000 users
  </TimeLineElement>
  <TimeLineElement title="Published in 'Nature Methods'" subtitle="February 2020">
    DOI: <a href="https://doi.org/10.1038/s41587-020-0439-x" class="font-normal text-gray-600 dark:text-gray-400">10.1038/s41587-020-0439-x</a>
  </TimeLineElement>
  <TimeLineElement title="Started project" subtitle="October 2018">
    Co-founded project with Alex Peltzer, after the 2018 Nextflow Meeting
  </TimeLineElement>
</div>
